### ISSUE TITLE
ANR on DefaultChannel$1.run DefaultChannel.java:514

### ISSUE THREAD

**Issue by drakeet** [microsoft/appcenter-sdk-android#1421]
Hi, we captured an ANR on `DefaultChannel$1.run DefaultChannel.java:514`, this is the relevant source code:

![image](https://user-images.githubusercontent.com/5214214/84337597-874efc00-abcc-11ea-9955-e4c8e2028b62.png)

### **Description**

stacktrace:

![image](https://user-images.githubusercontent.com/5214214/84337730-e3b21b80-abcc-11ea-8829-f3c32a2d5767.png)

### **Details**

1. Which SDK version are you using?
    - 3.2.1
2. Which OS version did you experience the issue on?
    - Android 5.1, 6.0, 7.1.2, 7.0
3. What device version did you see this error on?  Were you using an emulator or a physical device?
![image](https://user-images.githubusercontent.com/5214214/84337417-10196800-abcc-11ea-9aa3-c6cb063a0763.png)

4. What third party libraries are you using?
    - N/A
5. Please enable verbose logging for your app using `AppCenter.setLogLevel(Log.VERBOSE)` before your call to `AppCenter.start(...)` and include the logs here:

Sorry, it's a prod report, we cannot get the verbose logs.

**Comment by AnastasiaKubova** [comment]
Hi, @drakeet ! Thanks for getting in touch with us!
Could you please share the full crash report and the reproducing steps?

**Comment by drakeet** [comment]
@AnastasiaKubova Thanks for such a quick reply!
I only get the report on AppCenter with my ANR watchdog:
![image](https://user-images.githubusercontent.com/5214214/84352405-dce8d000-abef-11ea-99f9-45a6236f2903.png)

The report shows that 25 users have encountered the same problem, and I read all reports, all pointing to the same code:

![image](https://user-images.githubusercontent.com/5214214/84352555-26391f80-abf0-11ea-9296-1fa635802d5d.png)

**Comment by drakeet** [comment]
If @AnastasiaKubova you are not sure about the specific root cause, can you tell me if we can change this `sendLogs()` to be called in an asynchronous thread, and our application supports at least Android 5.0

**Comment by MatkovIvan** [comment]
Hi @drakeet,

`sendLogs` starts an asynchronous task, but the async task itself is triggered from the UI thread (but the work is done in a background thread).
The root cause is actually a `synchronized` with a state check happening before we trigger the async task that can block for a long time.
I created a PR (#1422) with removing `synchronized`, but it's still in progress since we probably can simplify this part because we bumped minimal version to 4.1 a while ago. So

> can you tell me if we can change this sendLogs() to be called in an asynchronous thread

this requires some future investigation. If it's dead code, we'll remove it
